ruby_on_rails/advanced_forms_and_activerecord/installing_psql.md

```diff
@@ -0,0 +1,339 @@
+### Introduction
+
+One of the great things about Rails is that it comes with some sensible defaults so that you can get started quickly. However, SQLite is typically not used for production. One major issue is that many hosting providers have ephemeral file systems which means your database can be deleted.
+
+For the apps written so far, it's probably not a big deal. As we progress to writing larger applications, though, it will become increasingly sensible to make sure that data persists even after a restart or hibernation.
+
+It is also sensible to make sure the application development environment is as close to the production environment as possible. This helps to avoid problems that stem from using different tools in different environments.
+
+### Lesson Overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+-  What an environment is and how it can impact our application.
+-  What environment variables are and how to utilize them.
+-  How to install PostgreSQL and set it up.
+-  How to set up a bare Rails application to use a specific database.
+
+### Environments and Pipelines
```

Comment:
> It feels like this environments and environment variables content could be in it's own lesson and expanded to discuss app specific env vars with Rails credentials.
> 
> Storing the database password in the bashrc is absolutely fine for this lesson since it's shared among many apps. But I'm a little worried leaners think all env vars should be stored in the bashrc when most of the env vars they will be scoped to individual app's.
> 
> This isn't a blocker though, I'd rather ship this and see if anyone actually runs into any issues or leave it as future enhancement.

Yes, I felt this as well writing the lesson, and in particular adding the knowledge checks really made this lesson feel a bit too broad. I know there is a suggestion discussion on the Discord about potentially giving learners a short introduction to the concepts of CI/CD. Maybe there's an future opportunity to build a little section that goes more in depth about environment variables alongside CI/CD.